migrations/1677667242101-EditHomeworkCreateAndEditPermissions.js

```diff
@@ -0,0 +1,94 @@
+const mongoose = require('mongoose');
+// eslint-disable-next-line no-unused-vars
+const { alert, error } = require('../src/logger');
+
+const { connect, close } = require('../src/utils/database');
+
+// use your own name for your model, otherwise other migrations may fail.
+// The third parameter is the actually relevent one for what collection to write to.
+const Roles = mongoose.model(
+	'role_202304011545',
+	new mongoose.Schema(
+		{
+			name: { type: String, required: true },
+			permissions: [{ type: String }],
+		},
+		{
+			timestamps: true,
+		}
+	),
+	'roles'
+);
+
+module.exports = {
+	up: async function up() {
+		await connect();
+
+		await Roles.updateOne(
+			{
+				name: 'teacher',
+			},
+			{
+				$addToSet: {
+					permissions: {
+						$each: ['HOMEWORK_CREATE', 'HOMEWORK_EDIT'],
+					},
+				},
+			}
+		).exec();
+		alert(`Permission HOMEWORK_CREATE and HOMEWORK_EDIT added to role teacher`);
+
+		await Roles.updateMany(
+			{
+				name: {
+					$in: ['user', 'courseStudent'],
+				},
+			},
+			{
+				$pull: {
+					permissions: {
+						$in: ['HOMEWORK_CREATE', 'HOMEWORK_EDIT'],
+					},
+				},
+			}
+		).exec();
+
+		alert(`Permission HOMEWORK_CREATE and HOMEWORK_EDIT removed from role user and courseStudent`);
+	},
+
+	down: async function down() {
+		await connect();
+
+		await Roles.updateMany(
+			{
+				name: {
+					$in: ['user', 'courseStudent'],
+				},
+			},
+			{
+				$addToSet: {
+					permissions: {
+						$each: ['HOMEWORK_CREATE', 'HOMEWORK_EDIT'],
+					},
+				},
+			}
+		).exec();
+		alert(`Permission HOMEWORK_CREATE and HOMEWORK_EDIT added to role user and courseStudent`);
+
+		await Roles.updateOne(
+			{
+				name: 'teacher',
+			},
+			{
+				$pull: {
+					permissions: {
+						$in: ['HOMEWORK_CREATE', 'HOMEWORK_EDIT'],
+					},
+				},
+			}
+		).exec();
+		alert(`Permission HOMEWORK_CREATE and HOMEWORK_EDIT removed from role teacher`);
```

Comment:
I think the down function is simply not intendend to be run, 
this is because it does not simply undo what the up did. 
I mean, in this case we want to run down function, this should restore previous data, but it does more: it deletes any some of those permissions, regardless if they were added from up function or were existing before

Reply:
So I should change it to something like this: https://github.com/hpi-schul-cloud/schulcloud-server/blob/088d903e4eaef6ccfad2e920c3a5b5769d25a2c3/migrations/1669628369773-add-submitted-and-graded-keys-to-submission.js#L58-L60

Reply:
@virgilchiriac Can you check if it is correct now?